part-spec/power.json

```diff
@@ -697,6 +697,516 @@
+          "$ref": "definitions.json#/unit"
+        },
+        "dimmingFrequencyMin": {
+          "description": "minimum dimming frequency of a device",
+          "comment": "units of hertz",
+          "$ref": "definitions.json#/unit"
+        },
+        "dimmingFrequencyMax": {
+          "description": "maximum dimming frequency of a device",
+          "comment": "units of hertz",
+          "$ref": "definitions.json#/unit"
+        },
+        "dimmingRatioTyp": {
+          "description": "typical dimming ratio of a device",
+          "type": "object",
+          "example": [
+          {"numerator": 20},
+          {"denominator": 1}
+          ]
+        },
+        "dimmingRatioMin": {
+          "description": "minimum dimming ratio of a device",
+          "type": "object",
+          "example": [
+          {"numerator": 10},
+          {"denominator": 1}
+          ]
+        },
+        "dimmingRatioMax": {
+          "description": "maximum dimming ratio of a device",
+          "type": "object",
+          "example": [
+          {"numerator": 40},
+          {"denominator": 1}
+          ]
+        },
+        "componentProtectionThresholds": {
+          "description": "Thermal and power supply protection thresholds of a device",
+          "type": "array",
+          "items": {
+            "$ref": "icProtection.json#/componentProtectionThresholds"
+          }
+        },
+        "package": {
+          "description": "package size of resistor",
+          "$ref": "definitions.json#/package"
+        },
+        "pins": {
+          "description": "array of pin objects with associated properties",
+          "type": "array",
+          "items": {
+            "$ref": "definitions.json#/pinSpec"
+          }
+        }
+      }
+    },
+    {
+      "title": "battery charger",
+      "type": "object",
+      "required": [
+        "componentID",
+        "batteryChargeCurrentMax"
+      ],
+      "properties": {
+        "componentID": {
+          "description": "common component identifying information, such as mpn",
+          "$ref": "definitions.json#/componentID"
+        },
+        "chargerType": {
+          "description": "battery charger type",
+          "examples": [
+            "linear",
+            "switching",
+            "direct"
+          ],
+          "type": "string"
+        },
+        "converterType": {
+          "description": "switching charger type",
+          "examples": [
+            "buck",
+            "boost",
+            "buck-boost",
+            "other"
+          ],
+          "type": "string"
+        },
+        "chargerTopology": {
+          "description": "type of battery charger topology (Narrow VDC vs Hybrid Power Boost)",
+          "enum": [
+            "nvdc",
+            "hpb"
+          ],
+          "type": "string"
+        },
+        "batteryConfig": {
+          "description": "battery configuration supported by the device",
```

Comment:
this should be an array since chargers can support multiple batter configs

Reply:
Also the number in series is generally more important than the number in parallel as I recall

Reply:
Yes, I'll change this and couple of other property types to array since multiple configs can be supported. As for the S & P, wanted to broaden it out so that it's not too centric to Notebooks.